### ISSUE TITLE
Crawl / manually curate / map Publications resources into individual metadata records based off of schema

### ISSUE THREAD

**Issue by flaneuse** [SuLab/outbreak.info-resources#25]
https://docs.google.com/spreadsheets/d/1FFyRhI5TeUb-B4t50HRZYF_XngRt_mfAIz82UQ3geFk/edit#gid=0

Need to check how often preprints from individual journals (Lancet, Nature, etc.) get incorporated into Pubmed index.  Don't want to be redundant, but also want to be as up-to-date as possible.

**Comment by andrewsu** [comment]
To start, I think harvesting from [LitCovid ](https://www.ncbi.nlm.nih.gov/research/coronavirus/) would be the easiest starting point.  My guess is that the PubMed/NCBI team is working hard on getting COVID-19-related literature into PubMed as quickly as possible, so I think we should old off on doing any direct scraping of journal sites.  Preprint servers like Biorxiv and Medrxiv would be more useful I think.  

So to summarize, I think the priority list should be 1) LitCovid, 2) Biorxiv / Medrxiv, and 3) journal sites.

(not seeing a biorxiv/medrxiv API, but maybe can use their rss feed and/or https://www.rxivist.org/docs)...

**Comment by flaneuse** [comment]
Makes sense.  

BioRxiv has a third-party API: https://api.rxivist.org/v1/papers?q=nCoV&timeframe=alltime. Little bit wonky; `&metric=downloads` gives different results. Unclear how frequently is updated. Could also be gotten via scraping https://www.biorxiv.org/search/ncov or [Twitter feeds](https://twitter.com/medrxivpreprint?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor).  Medrxiv searches seem to include Biorxiv but bio doesn't include med.  There's also Chemrxiv, but lower priority.

**Comment by flaneuse** [comment]
plan:
1. Pull data from LitCOVID/Pubmed: https://ftp.ncbi.nlm.nih.gov/pub/lu/LitCovid/. Use litcovid2BioCJSON.gz to get the IDs, and then use the [PubMed API](https://www.ncbi.nlm.nih.gov/home/develop/api/) to get the actual records
2. Scrape Biorxiv
3. Scripe Medrxiv
4. Align metadata with schema (#1 )

**Comment by gtsueng** [comment]
The mapping from PubMed/Medline to the schema can be found on the [worksheet](https://docs.google.com/spreadsheets/d/1BQAIT2Vw-I8SqWFArQP1dmj9gcv1lX7oNd-nTEOi9yY/edit#gid=0) in columns A and J

**Comment by flaneuse** [comment]
Additional information to grab from efetch:
example record (not COVID-related): https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=28841416&rettype=abstract

MeSH/OT terms --> `keywords`
abstract --> `abstract`
author affiliations --> `author.affiliation`
grants --> `funding`
referenceList --> `citedBy` 

@gtsueng has additional mapping of PubMed/Medline columns to our schema in columns A/J of the worksheet

**Comment by gtsueng** [comment]
Note that there are two methods for pulling the data from pubmed.  One method is to use the API which yields an xml file:
`https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=28841416&rettype=abstract`

The benefit of this method is as follows:
-Author given name and family name
-References (Citations)
-Specific grant information

The other method uses the biopython library to fetch and parse a medline record which treats it like a json file. Sample gist [here](https://gist.github.com/gtsueng/e2141866118c70a9c6b0cd6ccc5ec790).

The benefit of this method is as follows:
-json
-easy to map (the dictionary keys already mapped to schema fields in the worksheet: Publications, column J)
-PublicationType available as PT via this method
-Other terms aside also available (OT) aside from just MeSH (MH)

**Comment by gtsueng** [comment]
Abbreviated Journal titles may work better than the actual journal titles when it comes down to the visualization part. Since there are different ways to abbreviate journal titles, it may be better to use a dictionary for consistency. The NLM list of journal can be found [here](ftp://ftp.ncbi.nih.gov/pubmed/J_Medline.txt) and parsed into a dictionary. Once parsed, a manual review/edit could help to ensure that the most common abbreviations are rather than just the NLM abbreviations (which are by no means the de facto standard).

**Comment by flaneuse** [comment]
> Abbreviated Journal titles may work better than the actual journal titles when it comes down to the visualization part. Since there are different ways to abbreviate journal titles, it may be better to use a dictionary for consistency. The NLM list of journal can be found at ftp://ftp.ncbi.nih.gov/pubmed/J_Medline.txt and parsed into a dictionary. Once parsed, a manual review/edit could help to ensure that the most common abbreviations are rather than just the NLM abbreviations (which are by no means the de facto standard).

would propose having `journalName` be the full journal name (for instance `The New England journal of medicine`) and something like `journalNameAbbrevation` be `N. Engl. J. Med.`

**Comment by gtsueng** [comment]
Added into schema as suggested above. Still suggest populating from a single dictionary and caution against loading it from metadata from different sources since they may abbreviate journal titles differently.

**Comment by gtsueng** [comment]
> 
> referenceList --> `citedBy` (note: will need to pick an identifier to map to the other publications that cite this publication. ideally, this should refer to the `_id` we use to index these publications)
> 
@flaneuse In my mind the reference list should map to `isBasedOn` since those would be explicitly called out/referenced in the text and the `citedBy` would refer to other publications that cite this publication.

**Comment by flaneuse** [comment]
> > referenceList --> `citedBy` (note: will need to pick an identifier to map to the other publications that cite this publication. ideally, this should refer to the `_id` we use to index these publications)
> 
> @flaneuse In my mind the reference list should map to `isBasedOn` since those would be explicitly called out/referenced in the text and the `citedBy` would refer to other publications that cite this publication.

Oh, sorry, you're right; I'm backwards.  I thought these were the `citedBy` listed in the [PubMed entry](https://pubmed.ncbi.nlm.nih.gov/28841416/?from_single_result=28841416&expanded_search_query=28841416) -- I care more about those than the references they cite.  Wonder where these are hidden in the PubMed API...

**Comment by gtsueng** [comment]
I just assumed that they could query for those as long as all their other publications have references. In terms of related articles where the relationship is unclear, Medline will sometimes provide them as [AID] (associated ID). These are often related articles. Eg- if Publication A is a letter to the editor with regards to Publication B. An identifier for Publication B would be included as AID in the record for Publication A.  Unfortunately, AID is generic enough that they'll sometimes include the DOI for Publication A in the record for Publication A.

**Comment by flaneuse** [comment]
@marcodarko can you also track the LitCovid / BioRxiv / MedRxiv original url?  And maybe the source type (LitCovid, BioRxiv, or MedRxiv)?  

per @andrewsu on the front-end: "assuming we get that from litcovid, would be good to add a note and/or link.  Yes, probably somewhat redundant to do it on every page, but would rather err on the side of overattribution..."

I think we want to reserve `url` for the actual link to the paper (I don't want to get routed through PubMed when I just want to read the paper).  I'm not sure I see a natural schema.org way to capture this information... thoughts @gtsueng?  Proposal for your reaction (as an aside: we also need to decide on what we're using for canonical IDs; this will be used as the key to link between records, and also for the url to route on outbreak.info):
```
{ _id: "pmid32292904",
pmid: "32292904",
doi: "10.1016/j.xkme.2020.03.001",
url: "https://www.kidneymedicinejournal.org/article/S2590-0595(20)30049-2/pdf",
sameAs: {
name: "LitCovid",
url: "https://www.ncbi.nlm.nih.gov/research/coronavirus/publication/32292904"
}
}
```

**Comment by flaneuse** [comment]
> @marcodarko can you also track the LitCovid / BioRxiv / MedRxiv original url? And maybe the source type (LitCovid, BioRxiv, or MedRxiv)?
> 
> per @andrewsu on the front-end: "assuming we get that from litcovid, would be good to add a note and/or link. Yes, probably somewhat redundant to do it on every page, but would rather err on the side of overattribution..."
> 
> I think we want to reserve `url` for the actual link to the paper (I don't want to get routed through PubMed when I just want to read the paper). I'm not sure I see a natural schema.org way to capture this information... thoughts @gtsueng? Proposal for your reaction (as an aside: we also need to decide on what we're using for canonical IDs `_id`; this will be used as the key to link between records, and also for the url to route on outbreak.info):
> 
> ```
> { _id: "pmid32292904",
> pmid: "32292904",
> doi: "10.1016/j.xkme.2020.03.001",
> url: "https://www.kidneymedicinejournal.org/article/S2590-0595(20)30049-2/pdf",
> sameAs: {
> name: "LitCovid",
> url: "https://www.ncbi.nlm.nih.gov/research/coronavirus/publication/32292904"
> }
> }
> ```

update: creating new base class property `curatedBy` to store this info

**Comment by flaneuse** [comment]
Noting source of [BioRxiv data](https://connect.biorxiv.org/relate/content/181)

**Comment by flaneuse** [comment]
https://github.com/marcodarko/litcovid
https://github.com/marcodarko/biorxiv
